src/Features/Core/Portable/InlineHints/IInlineParameterNameHintsService.cs

```diff
@@ -16,6 +17,6 @@ namespace Microsoft.CodeAnalysis.InlineHints
     /// </summary>
     internal interface IInlineParameterNameHintsService : ILanguageService
     {
-        Task<ImmutableArray<InlineHint>> GetInlineHintsAsync(Document document, TextSpan textSpan, CancellationToken cancellationToken);
+        Task<ImmutableArray<InlineHint>> GetInlineHintsAsync(Document document, TextSpan textSpan, InlineParameterHintsOptions options, SymbolDescriptionOptions displayOptions, CancellationToken cancellationToken);
```

Comment:
can we just pass InlintHintsOptions here?

Reply:
Depends on how fine grained we want to be. InlintHintsOptions = parameter options + type options + display options. For this service type options are irrelevant, so when using this API separately it's odd that i'd need to specify options that are not applicable.

Reply:
that's fair.  but i'm also ok as these are really teh "General set of options for this feature area".  parts of the feature area need a slightly smaller subset.  but it still seems just generally clean and easy that in this feature-area we pass around that single represetnation.

i'd feel different if this api only needed *one* sub-portion.  but since it needs multiple, i'd rather just take the domain-options object so there's just a single thing passed along :)

Reply:
I guess the question is whether we want one options type per feature area vs one per service interface. I lean towards the latter since the correspondence is better defined.

I wonder why we need two different language services. Why not merge these interfaces into a single service with two methods?

Reply:
we def could do that.  i think i just liked the separation of concerns as there wasn't a lot of overlap at the end of the day and i didn't want some of the code using improper logic from the other.  for example our rules on how alt-f1 works for types is different for parameters. e.g. `alt-f1` for types will have the type literally replace the `var` in code.  nothing like hat exists for parametres.

Reply:
note: that the parameter part is entirely internal to roslyn c#/vb.  for the other languages we just expose a single top-level service type;

Reply:
```c#
    /// <summary>
    /// Gets inline hints for type locations.  This is an internal service only for C# and VB.  Use <see
    /// cref="IInlineHintsService"/> for other languages.
    /// </summary>
    internal interface IInlineParameterNameHintsService : ILanguageService
```

Reply:
>  i think i just liked the separation of concerns as there wasn't a lot of overlap at the end of the day

That was my understanding as well hence it felt like these should have different option types as well to maintain that separation.

Reply:
taht works for me :)